src/components/ReleaseService/ReleasePlan/TriggerRelease/form-utils.ts

```diff
@@ -0,0 +1,98 @@
+import { k8sCreateResource } from '@openshift/dynamic-plugin-sdk-utils';
+import * as yup from 'yup';
+import {
+  MAX_RESOURCE_NAME_LENGTH,
+  resourceNameRegex,
+  RESOURCE_NAME_LENGTH_ERROR_MSG,
+  RESOURCE_NAME_REGEX_MSG,
+} from '../../../../components/ImportForm/utils/validation-utils';
+import { ReleaseGroupVersionKind, ReleaseModel } from '../../../../models';
+import { ReleaseKind, ReleasePlanKind } from '../../../../types/coreBuildService';
+
+export enum ReleasePipelineLocation {
+  current,
+  target,
+}
+
+export type TriggerReleaseFormValues = {
+  releasePlan: string;
+  snapshot: string;
+  synopsis: string;
+  topic: string;
+  description?: string;
+  solution?: string;
+  references: string;
+  issues?: any[];
+  cves?: any[];
+  labels?: { key: string; value: string }[];
+};
+
+export const triggerReleaseFormSchema = yup.object({
+  releasePlan: yup
+    .string()
+    .matches(resourceNameRegex, RESOURCE_NAME_REGEX_MSG)
+    .max(MAX_RESOURCE_NAME_LENGTH, RESOURCE_NAME_LENGTH_ERROR_MSG)
+    .required('Required'),
+  snapshot: yup
+    .string()
+    .matches(resourceNameRegex, RESOURCE_NAME_REGEX_MSG)
+    .max(MAX_RESOURCE_NAME_LENGTH, RESOURCE_NAME_LENGTH_ERROR_MSG)
+    .required('Required'),
+});
+
+export const createRelease = async (
+  values: TriggerReleaseFormValues,
+  namespace: string,
+  releasePlan: ReleasePlanKind,
+) => {
+  const {
+    releasePlan: rp,
+    snapshot,
+    cves,
+    topic,
+    labels: labelPairs,
+    description,
+    solution,
+    issues,
+    synopsis,
+  } = values;
+
+  const labels = labelPairs
+    .filter((l) => !!l.key)
+    .reduce((acc, o) => ({ ...acc, [o.key]: o.value }), {} as Record<string, string>);
+
+  const resource: ReleaseKind = {
+    apiVersion: `${ReleaseGroupVersionKind.group}/${ReleaseGroupVersionKind.version}`,
+    kind: ReleaseGroupVersionKind.kind,
+    metadata: {
+      generateName: rp,
+      namespace,
+      labels: {
+        ...releasePlan?.metadata?.labels,
+        ...labels,
+      },
+      annotations: { ...releasePlan?.metadata?.annotations },
+    },
+    spec: {
+      releasePlan: rp,
+      snapshot,
+      data: {
+        advisory: {
+          cves,
+          issues,
```

Comment:
talk to errata team to figure out if bugs/cves should be under references

Reply:
no update yet from errata team. I am using fields as per the latest conversation. I see the object is being created with the correct values in releaseNotes.
Maybe the errata team should remap the values in their API end or file a bug with all the changes they request as there is no document/link provided thus far